ginga/canvas/types/plots.py

```diff
@@ -0,0 +1,832 @@
+        super(YAxis, self).__init__(title=title, num_labels=num_labels,
+                                    font=font, fontsize=fontsize)
+
+        self.kind = 'axis_y'
+        self.title_wd = 0
+        self.txt_wd = 0
+        self.txt_ht = 0
+        self.pad_px = 4
+
+    def register_decor(self, aide):
+        self.aide = aide
+
+        # add Y grid
+        self.grid = Bunch.Bunch()
+        for i in range(self.num_labels):
+            self.grid[i] = aide.dc.Line(0, 0, 0, 0, color=aide.grid_fg,
+                                        linestyle='dash', linewidth=1,
+                                        alpha=self.grid_alpha,
+                                        coord='window')
+            self.objects.append(self.grid[i])
+
+        # bg for RHS Y axis labels
+        self.axis_bg = aide.dc.Rectangle(0, 0, 100, 100, color=aide.norm_bg,
+                                         fill=True, fillcolor=aide.axis_bg,
+                                         coord='window')
+        self.objects.append(self.axis_bg)
+
+        # bg for LHS Y axis title
+        self.axis_bg2 = aide.dc.Rectangle(0, 0, 100, 100, color=aide.norm_bg,
+                                          fill=True, fillcolor=aide.axis_bg,
+                                          coord='window')
+        self.objects.append(self.axis_bg2)
+
+        # Y grid (tick) labels
+        self.lbls = Bunch.Bunch()
+        for i in range(self.num_labels):
+            self.lbls[i] = aide.dc.Text(0, 0, text='', color='black',
+                                        font=self.font,
+                                        fontsize=self.fontsize,
+                                        coord='window')
+            self.objects.append(self.lbls[i])
+
+        # Y title
+        self._title = aide.dc.Text(0, 0, text=self.title, color='black',
+                                   font=self.font,
+                                   fontsize=self.fontsize,
+                                   alpha=0.0,
+                                   rot_deg=90.0,
+                                   coord='window')
+        self.objects.append(self._title)
+
+    def update_elements(self, viewer):
+        """This method is called if the plot is set with new points,
+        or is scaled or panned with existing points.
+
+        Update the YAxis labels to reflect the new values and/or pan/scale.
+        """
+        # set Y labels/grid as needed
+        for i in range(self.num_labels):
+            lbl = self.lbls[i]
+            # get data coord equivalents
+            x, y = self.get_data_xy(viewer, (lbl.x, lbl.y))
+            lbl.text = self.format_value(y)
+
+    def update_bbox(self, viewer, dims):
+        """This method is called if the viewer's window is resized.
+
+        Update all the YAxis elements to reflect the new dimensions.
+        """
+        title = self.get_title()
+        self._title.text = title if title is not None else '555.55'
```

Comment:
Same question about `'555.55'` here.

Reply:
Just something to calculate the pixel size of text height. Not sure why I chose that particular value, other than I thought '5' kind of represents a typical digit size. This value is not shown if the user did not actually set a title, just used for the pixel size calculation.